### PR TITLE
support for `with socket.create_connection`

### PR DESCRIPTION
**Pull request by alicebob**
That would solve #1391 

Support for `with socket.create_connection` was added here: https://bugs.python.org/issue9794 in commit b383dbb45e62054e3a73792b7e903178bee18aa8 in cpython.

### CODE REVIEW

File: stdlib/2and3/socket.pyi [hunk 1]
```diff
@@ -505,6 +505,10 @@ class socket:
         def __init__(self, family: int = ..., type: int = ...,
                      proto: int = ..., fileno: Optional[int] = ...) -> None: ...
 
+    if sys.version_info >= (3, 2):
+        def __enter__(self) -> socket: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/socket.pyi]: Strictly speaking this should be `def __enter__(self: _SelfT) -> _SelfT: ...`, where `_SelfT` is a TypeVar bound to socket. This way, the right type will be inferred if somebody uses `with` on a subclass of socket. Not sure how common it is to subclass `socket.socket`, but better to be safe.

**Comment by alicebob** [comment]
Added that. Tested by hand (don't how where to add tests for this).

The new `_SelfT` needs the `socket` class, so I added it just where it's needed. If this is fine I'll squash it.

**Comment by JelleZijlstra** [comment]
Thanks! Looks like you have a small lint error and pytype doesn't like defining typevars in the class scope; could you fix that?

**Comment by alicebob** [comment]
Now it's the same as in stdlib/3/nntplib.py, which is the only other place I could find which defines `__enter__` with proper classes.

**Comment by alicebob** [comment]
Build is happy, @JelleZijlstra

**Comment by JelleZijlstra** [comment]
Thanks!

**Comment by alicebob** [comment]
Thanks for the help. Would you be interested in a PR which adds typevars to more stdlib __enter__ methods?

**Comment by JelleZijlstra** [comment]
Yes, that would be an improvement. It's not necessarily the highest priority thing though.

**Pull request merged**